### PR TITLE
Add Linting section

### LINKED ISSUE #63: Recommended Linters?
I love the uber-go guide! Thanks for sharing your accumulated experience. While I am aware of [awesome-go-linters](https://github.com/golangci/awesome-go-linters), I wonder if there are any specific recommended linters that compliment this guide.

Alternatively, are there any things in this guide that you *wish* there was a linter for? We at Khan Academy are contributing to some existing ones and writing a few in house ones as we increase our adoption.

### PR DESCRIPTION
**Pull request by mway**
Fixes #63.

### CODE REVIEW

**Comment by fungilation** [comment]
Can this be merged? I'm looking for recommended golangci config, and this adds to the style guide with something to actually run as CI in practice.

**Comment by mway** [comment]
> Can this be merged? I'm looking for recommended golangci config, and this adds to the style guide with something to actually run as CI in practice.

Thanks for your interest! We're in the process of discussing some specifics internally - once that's done, we'll update and land this so there's common guidance.

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Refer to golangci-lint's example config file for more options and information:
+# https://github.com/golangci/golangci-lint/blob/master/.golangci.example.yml
+
+run:
+  timeout: 5m
+  issues-exit-code: 1
+  tests: true
+  skip-dirs-use-default: true
```
Comment (abhinav) [inline .golangci.yml:8]: can delete these since these are the defaults.
```suggestion
```

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  modules-download-mode: readonly
+
+output:
+  format: colored-line-number
+  print-issued-lines: true
+  print-linter-name: true
+
```
Comment (abhinav) [inline .golangci.yml:15]: can delete because this is the default.
```suggestion
```

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    - golint
+    - govet
+    - staticcheck
+
+issues:
+  exclude-use-default: false
+  max-issues-per-linter: 0
+  max-same-issues: 0
+  new: false
+  new-from-rev: HEAD~
```
Comment (abhinav) [inline .golangci.yml:29]: Can delete. This is the default.

```suggestion
```

**Review by abhinav — APPROVED** [review]
lgtm with minor comments

CC @prashantv

**Pull request merged**